src/Network/Wai/Session/PostgreSQL.hs

```diff
@@ -279,11 +269,7 @@ writer pool key sessionPgId k v = do
     let k' = Binary $ encode k
         v' = Binary $ encode v
     liftIO $ withPostgreSQLConn pool $ \conn ->
-        withTransaction conn $ do
-            res <- query conn qryLookupSession''' (sessionPgId, k') :: IO [Only Int64]
-            case res of
-                [Only id]   -> void $ execute conn qryUpdateSessionEntry (v', sessionPgId, k')
-                _           -> void $ execute conn qryCreateSessionEntry (sessionPgId, k', v')
```

Comment:
I don't think this code could have possibly worked, regardless of isolation level.  On concurrent inserts, depending on the isolation level, it will either fail with a constraint violation (`read committed`), or the transaction will not commit (`repeatable read`, `serializable`).